tests/core/contracts/test_contract_call_interface.py

```diff
@@ -902,6 +928,115 @@ def test_call_tuple_contract(tuple_contract, method_input, expected):
+                            "0xf35C0784794F3Cd935F5754d3a0EbcE95bEf851e",
+                        ],
+                    },
+                ],
+            },
+            (
+                123,
+                [1, 2],
+                [
+                    (
+                        234,
+                        [True, False],
+                        [
+                            "0x4AD7E79d88650B01EEA2B1f069f01EE9db343d5c",
+                            "0xfdF1946A9b40245224488F1a36f4A9ed4844a523",
+                            "0xfdF1946A9b40245224488F1a36f4A9ed4844a523",
+                        ],
+                    ),
+                    (
+                        345,
+                        [False, False],
+                        [
+                            "0xefd1FF70c185A1C0b125939815225199079096Ee",
+                            "0xf35C0784794F3Cd935F5754d3a0EbcE95bEf851e",
+                        ],
+                    ),
+                ],
+            ),
+        ),
+        (
+            (
+                123,
+                [1, 2],
+                [
+                    (
+                        234,
+                        [True, False],
+                        [
+                            "0x4AD7E79d88650B01EEA2B1f069f01EE9db343d5c",
+                            "0xfdF1946A9b40245224488F1a36f4A9ed4844a523",
+                            "0xfdF1946A9b40245224488F1a36f4A9ed4844a523",
+                        ],
+                    ),
+                    (
+                        345,
+                        [False, False],
+                        [
+                            "0xefd1FF70c185A1C0b125939815225199079096Ee",
+                            "0xf35C0784794F3Cd935F5754d3a0EbcE95bEf851e",
+                        ],
+                    ),
+                ],
+            ),
+            (
+                123,
+                [1, 2],
+                [
+                    (
+                        234,
+                        [True, False],
+                        [
+                            "0x4AD7E79d88650B01EEA2B1f069f01EE9db343d5c",
+                            "0xfdF1946A9b40245224488F1a36f4A9ed4844a523",
+                            "0xfdF1946A9b40245224488F1a36f4A9ed4844a523",
+                        ],
+                    ),
+                    (
+                        345,
+                        [False, False],
+                        [
+                            "0xefd1FF70c185A1C0b125939815225199079096Ee",
+                            "0xf35C0784794F3Cd935F5754d3a0EbcE95bEf851e",
+                        ],
+                    ),
+                ],
+            ),
+        ),
+    ),
+)
+def test_call_tuple_contract_with_decode_tuples_set(
+    tuple_contract_with_decode_tuples, method_input, expected
+):
+    result = tuple_contract_with_decode_tuples.functions.method(method_input).call()
+    assert result == expected
```

Comment:
In general, I think these tests would be clearer if we were also checking for that the `type(expected)` is equal to the namedtuple class name, and maybe even checking the `__repr__` to make sure everything gets assigned like it should. It took me a while to figure out why an apparently "normal" tuple was returned here ha.

Reply:
I agree. Changed some naming and added several more checks. I wasn't sure how to check against the full namedtuple form, but using `__repr__` works great!